src/sentry/sentry_apps/tasks/sentry_apps.py

```diff
@@ -146,7 +151,41 @@ def send_alert_event(
         return
     (install,) = installations
 
-    event_context = _webhook_event_data(event, group.id, project.id)
+    nodedata = nodestore.backend.get(
+        BaseEvent.generate_node_id(project_id=project.id, event_id=instance_id)
+    )
+
+    if not nodedata:
+        extra = {
+            "event_id": instance_id,
+            "occurrence_id": occurrence_id,
+            "rule": rule,
+            "sentry_app": sentry_app.slug,
+            "group_id": group_id,
+        }
+        logger.info("send_alert_event.missing_event", extra=extra)
+        return
+
+    occurrence = None
+    if occurrence_id:
+        occurrence = IssueOccurrence.fetch(occurrence_id, project_id=project.id)
```

Comment:
note: we don't actually use/expose any of the occurrence data currently in the eventual webhook payload. See [_webhook_event_data](https://github.com/getsentry/sentry/blob/master/src/sentry/sentry_apps/tasks/sentry_apps.py#L80) where [as_dict](https://github.com/getsentry/sentry/blob/master/src/sentry/eventstore/models.py#L486) only pulls out attributes from the `data` attribute which is the same for both `Event` & `GroupEvent` objects. I don't know if we're not supposed to expose the occurrence info (?) but if we aren't we can get rid of any occurrence fetches & potentially change the typing to be only `Event`

Reply:
As part of implementing the issue platform we just passed around the group event with occurrence everywhere since we wanted consistency. It's possible we've used this in places where we don't need to, or that the use got lost over time and we didn't remove it.

I'd guess it's probably just related to https://github.com/getsentry/sentry/blob/15cb8dee5b46af99f7390882d5fe39fb7b0ce3c7/src/sentry/sentry_apps/tasks/sentry_apps.py#L241-L243.

But if we're not using the occurrence here then feel free to remove it

Reply:
For context this webhook/task (`send_alert_event`) fires whenever any corresponding alert (error, metric, issue etc) is triggered. So I'd assume we'd want to send the occurrence data in the webhook. Idk maybe `_webhook_event_data` was written before `GroupEvents`/`occurrences` were a thing 🤷‍♀️ .